x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/allocation/decider/HasFrozenCacheAllocationDecider.java

```diff
@@ -83,7 +90,8 @@ private Decision canAllocateToNode(IndexMetadata indexMetadata, DiscoveryNode di
             case HAS_CACHE -> HAS_FROZEN_CACHE;
             case NO_CACHE -> NO_FROZEN_CACHE;
             case FAILED -> UNKNOWN_FROZEN_CACHE;
-            default -> STILL_FETCHING;
+            case FETCHING -> STILL_FETCHING;
```

Comment:
I've only briefly looked at this code area, but I'm wondering why we'd use `THROTTLE` instead of `NO`. I would expect the `FrozenCacheService` to make a `reroute  when it receives the update, but it doesn't look like that happens. So `THROTTLE` looks like a hack. Could we make that `reroute` happen in a new `FrozenCacheService#clusterChanged`?

Reply:
> I left it unchanged for non-simulation since that is not important to us and less change.

So it's safe to make it always NO, but you're trying to minimize the change?

Reply:
The split / explicit switches per enum is all good to me, I'm more hesitant about not fully changing over to NO. I think a remaining throttle case leaves the code in a confusing state: it's technical debt.

I'm good with pushing the change as is.